filters/include/pcl/filters/filter_indices.h

```diff
@@ -157,14 +184,26 @@ namespace pcl
         user_filter_value_ = value;
       }
 
-    protected:
 
-      using Filter<PointT>::initCompute;
-      using Filter<PointT>::deinitCompute;
-      using Filter<PointT>::input_;
-      using Filter<PointT>::removed_indices_;
+      /** \brief overloaded filter method with specified executor.
+       *
+       * The implementation needs to set output.{points, width, height, is_dense}.
+       *
+       * \param[int] exec the executor to run the filter using
+       * \param[out] output the resultant filtered point cloud
+       */
+      template <typename Executor>
+      void
+      applyFilter (const Executor &exec, PointCloud &output);
```

Comment:
This should not be in public scope. It's supposed to be only called between `initCompute` and `deinitCompute` calls. What happened?

Reply:
The base class `Filter` needs to call it that's why it is public. But I think it's possible to access protected methods in CRTP. Should I go ahead with that approach?

Reply:
What was the conclusion here?

Reply:
To make the Base class as a friend of the Derived (db61393)